app/code/Apruve/Payment/Model/PaymentMethod.php

```diff
@@ -7,248 +7,300 @@ class PaymentMethod extends \Magento\Payment\Model\Method\AbstractMethod
-        
-        $payment->setTransactionId($token)->setIsTransactionClosed(0);
-        
-        // If Reserved Order ID is not correct
-        $order = $payment->getOrder();
-        if($response->merchant_order_id != $order->getIncrementId()); {
-            $this->_updateMerchantID($token, $order->getIncrementId());
-        }
-        
+    protected function _isAdmin($store = null)
+    {
+        /** @var \Magento\Framework\ObjectManagerInterface $om */
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+        /** @var \Magento\Framework\App\State $state */
+        $state = $objectManager->get('Magento\Framework\App\State');
+
+        return 'adminhtml' === $state->getAreaCode();
     }
-    
-    public function capture(\Magento\Payment\Model\InfoInterface $payment, $amount) {
+
+    public function authorize(\Magento\Payment\Model\InfoInterface $payment, $amount)
+    {
         $order = $payment->getOrder();
-        $data = $payment->getAdditionalInformation();
-        $token = $data['aid'];
+
+        # Authorize the order
+        $corporateAccount = $this->_getCorporateAccount($order->getCustomerEmail());
+        $buyer            = $this->_getBuyer($corporateAccount, $order->getCustomerEmail());
+        if (empty($buyer->id)) {
+            throw new \Magento\Framework\Validator\Exception(__('Cannot find Apruve corporate account for that customer'));
+        }
 
         /**Validate*/
         if ($amount <= 0) {
             throw new \Magento\Framework\Validator\Exception(__('Invalid amount for capture.'));
         }
 
-        /**Create Invoice*/
-        $invoices = $order->getInvoiceCollection();
-        foreach ($invoices as $invoice) {
-            $invoiceId = $invoice->getId();
-        }
-        
-        $data = array();
-        $data['amount_cents'] = ($order->getPayment()->getData('amount_ordered')) * 100;
-        $data['currency'] = $order->getData('base_currency_code');
-        $data['shipping_cents'] = $order->getData('shipping_amount') * 100;
-        $data['tax_cents'] = $order->getData('tax_amount') * 100;
-        $data['merchant_notes'] = '';
-        $data['merchant_invoice_id'] = $invoiceId;
-        $data['due_at'] = '';
-        $data['invoice_items'] = [];
-        $data['issue_on_create'] = 'true';
-        
+        $data                        = [];
+        $data['merchant_id']         = $this->_getMerchantKey();
+        $data['merchant_order_id']   = $order->getIncrementId();
+        $data['shopper_id']          = $buyer->id;
+        $data['amount_cents']        = ( $order->getPayment()->getData('amount_ordered') ) * 100;
+        $data['currency']            = $order->getData('base_currency_code');
+        $data['shipping_cents']      = $order->getData('shipping_amount') * 100;
+        $data['tax_cents']           = $order->getData('tax_amount') * 100;
+        $data['merchant_notes']      = '';
+        $data['merchant_invoice_id'] = 'blah!';
```

Comment:
I think we'd prefer this stay unset or blank if there's no Magento internal invoice ID available here.